conda/gateways/disk/lock.py

```diff
@@ -30,13 +31,16 @@ def _lock_noop(fd):
     def _lock_impl(fd):  # type: ignore
         tell = fd.tell()
         fd.seek(LOCK_BYTE)
-        msvcrt.locking(fd.fileno(), msvcrt.LK_LOCK, 1)  # type: ignore
         try:
-            fd.seek(tell)
-            yield
-        finally:
-            fd.seek(LOCK_BYTE)
-            msvcrt.locking(fd.fileno(), msvcrt.LK_UNLCK, 1)  # type: ignore
+            msvcrt.locking(fd.fileno(), msvcrt.LK_LOCK, 1)  # type: ignore
+            try:
+                fd.seek(tell)
+                yield
+            finally:
+                fd.seek(LOCK_BYTE)
+                msvcrt.locking(fd.fileno(), msvcrt.LK_UNLCK, 1)  # type: ignore
+        except LockError as e:
+            print("Failed to acquire lock.", e)
```

Comment:
This exception will never happen

`msvcrt.locking` does not raise a `LockError`, it raises an `OSError`